blog/views.py

```diff
@@ -1,117 +1,127 @@
-        form = PostForm(request.POST, instance=post)
-        if form.is_valid():
-            post = form.save(commit=False)
-            post.author = request.user
-            post.save()
-            return redirect('blog.views.post_detail', pk=post.pk)
-    else:
-        form = PostForm(instance=post)
-    return render(request, 'blog/post_edit.html', {'form': form})
-
-
-@login_required
-def post_draft_list(request):
-    """Unpublished posts."""
-    posts = (
-        Post.objects
-        .filter(published_date__isnull=True)
-        .order_by('created_date')
-    )
-    return render(request, 'blog/post_draft_list.html', {'posts': posts})
-
-
-@login_required
-def post_publish(request, pk):
-    """Publish post."""
-    post = get_object_or_404(Post, pk=pk)
-    post.publish()
-    return redirect('post_detail', pk=pk)
-
-
-@login_required
-def post_remove(request, pk):
-    """Delete post."""
-    post = get_object_or_404(Post, pk=pk)
-    post.delete()
-    return redirect('post_list')
-
-
-def add_comment(request, pk):
-    """Add comment to post."""
-    post = get_object_or_404(Post, pk=pk)
-    if request.method == 'POST':
-        form = CommentForm(request.POST)
-        if form.is_valid():
-            comment = form.save(commit=False)
-            comment.post = post
-            comment.save()
-            return redirect('post_detail', pk=post.pk)
-    else:
-        form = CommentForm()
-    return render(request, 'blog/add_comment.html', {'form': form})
-
-
-@login_required
-def comment_approve(request, pk):
-    """Approve (publish) comment."""
-    comment = get_object_or_404(Comment, pk=pk)
-    comment.approve()
-    return redirect('post_detail', pk=comment.post.pk)
-
-
-@login_required
-def comment_remove(request, pk):
-    """Delete comment."""
-    comment = get_object_or_404(Comment, pk=pk)
-    post_pk = comment.post.pk
-    comment.delete()
-    return redirect('post_detail', pk=post_pk)
+class Protected(View):
+    """Deny access to view for unauthorized users."""
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        """Call base method with login_required extension."""
+        return super().dispatch(*args, **kwargs)
+
+
+class PostListView(ListView):
+    """View for list of all published posts."""
+
+    context_object_name = 'posts'
+    template_name = 'blog/post_list.html'
+
+    def get_queryset(self):
+        """Return already published posts."""
+        return Post.objects.filter(published_date__lte=timezone.now()).order_by('published_date')
+
+
+class PostDetailView(DetailView):
+    """View with detail info about post."""
+
+    model = Post
+    template_name = 'blog/post_detail.html'
+
+
+class NewPost(CreateView, Protected):
+    """View for creating new post."""
+
+    form_class = PostForm
+    template_name = 'blog/post_edit.html'
+
+    def form_valid(self, form):
+        """Pass request.user to model."""
+        form.save(self.request.user)
```

Comment:
No, `form.save()` is already being called after `form_valid()` by the base view's logic. I think, you probably should only modify the instance like `form.instance.author = self.request.user` and that's it:
https://docs.djangoproject.com/en/2.0/topics/class-based-views/generic-editing/#models-and-request-user